src/utils/daos_hdlr.c

```diff
@@ -1922,7 +1922,8 @@ file_write(struct cmd_args_s *ap, struct file_dfs *file_dfs,
 
 	if (file_dfs->type == POSIX) {
 		*size = write(file_dfs->fd, buf, *size);
-		if (*size < 0)
+		/* cast to fix compiler warning about comparison of unsigned type */
+		if ((int)*size < 0)
```

Comment:
This should be cast to the type returned from write
https://man7.org/linux/man-pages/man2/write.2.html
```suggestion
		/* cast to fix compiler warning about comparison of unsigned type */
		if ((ssize_t)*size < 0)
```

Reply:
Actually, daos_size_t is unsigned
```
typedef uint64_t      daos_size_t
```
So I think the ways dfs_sys sets -1 is incorrect, right?

Reply:
This is a larger issue than this PR - I'm going to create a ticket with the proper fix

Reply:
@daltonbohning yeah, thanks. I updated the file_read/write calls to use ssize_t

Reply:
@dsikich Thanks. LGTM. As we discussed, either way we are going to be limited in some way by ssize_t, since that is what the POSIX functions use. But this is okay for now since we don't pass sizes that would overflow the positive range of ssize_t. Maybe that could change in the future if the chunk size is made configurable by the user, at which time we could re-address this and make changes if necessary.
Also, I created https://jira.hpdd.intel.com/browse/DAOS-8293 for the dfs_sys part of this issue